File: csharp/src/Behavioral/Interpreter/DvdTitleActorExpression.cs
//DvdTitleActorExpression.java - Four Of Four Terminal Expressions

namespace Behavioral.Interpreter {

using System;
using System.Collections.Generic;


public class DvdTitleActorExpression : DvdAbstractExpression {  
   string title;
   
   public DvdTitleActorExpression(string title) {
       this.title = title;
   }
    
   override public string interpret(DvdInterpreterContext dvdInterpreterContext) 
	{
		List<string> titlesAndActors = dvdInterpreterContext.getTitlesForActor(title);
		return string.Join(", ",titlesAndActors.ToArray());
   }
}

}

File: go/behavioral/interpreter/DvdTitleActorExpression.go
package interpreter

import (
	"strings"
)

type DvdTitleActorExpression struct { // implements Interpretable
   title string
}

func NewDvdTitleActorExpression(title string) *DvdTitleActorExpression {
	return &DvdTitleActorExpression{ title }
}
    
func (this *DvdTitleActorExpression) Interpret (ctx *DvdInterpreterContext) string {
	if this.title == "" {
		return ""
	}
	return strings.Join(ctx.GetTitlesForActor( this.title ),", ")
}
